File: src/com/company/BinaryTreeLevelOrderTraversal.java
package com.company;


import java.util.ArrayList;
import java.util.LinkedList;
import java.util.List;
import java.util.Queue;

public class BinaryTreeLevelOrderTraversal {

    public static void main(String[] args) {
        var ret = new Solution().levelOrder(TreeNode.ROOT);
        System.out.println(ret);
    }
}

class Solution {
    public List<List<Integer>> levelOrder(TreeNode root) {
        var result = new LinkedList<List<Integer>>();
        if (root == null)
            return result;

        Queue<TreeNode> queue = new LinkedList<>();
        queue.add(root);
        while (!queue.isEmpty()) {
            int n = queue.size();

            List<Integer> level = new ArrayList<>();
            result.add(level);
            for (int i = 0; i < n; i++) {
                TreeNode node = queue.poll();
                level.add(node.val);
                if (node.left != null) queue.add(node.left);
                if (node.right != null) queue.add(node.right);
            }

        }

        return result;
    }
}

File: javascript/tree/BinaryTreeLevelOrderTraversal.js
import { TreeNode } from "./TreeNode.js";

const root = new TreeNode(
    1,
    new TreeNode(
        2,
        new TreeNode(4),
        new TreeNode(5)
    ),
    new TreeNode(
        3,
        new TreeNode(6),
        new TreeNode(7)
    )
);

class Solution {
    /**
     * @param {TreeNode} root
     * @return {number[][]}
     */
    levelOrder(root) {
        const result = [];
        if (!root) return result;

        const queue = [];

        queue.push(root);

        while (queue.length) {

            const n = queue.length;
            const levelResult = [];
            for (let i = 0; i < n; i++) {


                const node = queue.shift();
                levelResult.push(node.val);

                if (node.left) queue.push(node.left);
                if (node.right) queue.push(node.right);
            }
            result.push(levelResult);
        }

        return result;
    }
}

console.log(new Solution().levelOrder(root));